classes/submission/PKPSubmissionFileDAO.inc.php

```diff
@@ -0,0 +1,343 @@
+			->where('file_id', '=', $submissionFile->getData('fileId'))
+			->exists();
+
+		if (!$hasFileId) {
+			Capsule::table('submission_file_revisions')->insert([
+				'submission_file_id' => $submissionFile->getId(),
+				'file_id' => $submissionFile->getData('fileId'),
+			]);
+		}
+	}
+
+	/**
+	 * @copydoc SchemaDAO::deleteById()
+	 */
+	public function deleteById($submissionFileId) {
+		Capsule::table('submission_file_revisions')
+			->where('submission_file_id', '=', $submissionFileId)
+			->delete();
+
+		Capsule::table('review_round_files')
+			->where('submission_file_id', '=', $submissionFileId)
+			->delete();
+
+		parent::deleteById($submissionFileId);
+	}
+
+
+	//
+	// Public methods
+	//
+	/**
+	 * Retrieve file by public file ID
+	 * @param $pubIdType string One of the NLM pub-id-type values or
+	 * 'other::something' if not part of the official NLM list
+	 * (see <http://dtd.nlm.nih.gov/publishing/tag-library/n-4zh0.html>).
+	 * @param $pubId string
+	 * @param $submissionId int optional
+	 * @param $contextId int optional
+	 * @return SubmissionFile|null
+	 */
+	function getByPubId($pubIdType, $pubId, $submissionId = null, $contextId = null) {
+		if (empty($pubId)) {
+			return null;
+		}
+
+		$submissionFileId = Capsule::table('submission_files as sf')
+			->leftJoin('submission_file_settings as sfs', 'sfs.submission_file_id', '=' , 'sf.submission_file_id')
+			->where('sf.submission_id', '=', $submissionId)
+			->where(function($q) use ($pubIdType, $pubId) {
+				$q->where('sfs.setting_name', '=', 'pub-id::' . $pubIdType);
+				$q->where('sfs.setting_value', '=', $pubId);
+			})
+			->value('sf.submission_file_id');
+
+		if (empty($submissionFileId)) {
+			return null;
+		}
+
+		$submissionFile = Services::get('submissionFile')->get($submissionFileId);
+
+		if ($submissionFile->getData('fileStage') !== SUBMISSION_FILE_PROOF) {
+			return $submissionFile;
+		}
+
+		return null;
+	}
+
+	/**
+	 * Retrieve file by public ID or submissionFileId
+	 *
+	 * @param string|int $bestId Publisher id or submissionFileId
+	 * @param int $submissionId
+	 * @return SubmissionFile|null
+	 */
+	function getByBestId($bestId, $submissionId) {
+		$submissionFile = null;
+		if ($bestId != '') $submissionFile = $this->getByPubId('publisher-id', $bestId, $submissionId, null);
+		if (!isset($submissionFile)) {
+			$submissionFile = Services::get('submissionFile')->get($bestId);
+		}
+		if ($submissionFile && in_array($submissionFile->getData('fileStage'), [SUBMISSION_FILE_PROOF, SUBMISSION_FILE_DEPENDENT])) {
+			return $submissionFile;
+		}
+		return null;
+	}
+
+	/**
+	 * Assign file to a review round.
+	 * @param $submissionFileId int The file to be assigned.
+	 * @param $reviewRound ReviewRound
+	 */
+	function assignRevisionToReviewRound($submissionFileId, $reviewRound) {
+
+		// Avoid duplication errors -- clear out any existing entries
+		$this->deleteReviewRoundAssignment($submissionFileId);
+
+		return $this->update(
```

Comment:
Lots of our DAO functions return the `parent::update(...)` return value -- which is actually the success/failure status of the query, not e.g. the number of affected rows. This will change when we go to the Laravel toolset, which does want to return the number of affected rows (and which developers often expect). And going forward Exceptions should be preferred to `nulls`/`false` returns for error conditions. I'd suggest not returning anything at all until this sorts itself out, particularly since there's no `@return` documented here and I don't think any calling code expects a return.